pinot-common/src/main/java/org/apache/pinot/pql/parsers/PinotQuery2BrokerRequestConverter.java

```diff
@@ -223,38 +226,66 @@ private String standardizeFunction(Function functionCall) {
 
   private AggregationInfo buildAggregationInfo(Function function) {
     List<Expression> operands = function.getOperands();
-    if (operands == null || operands.size() != 1) {
-      throw new Pql2CompilationException(
-          "Aggregation function" + function.getOperator() + " expects 1 argument. found: " + operands);
+    if (operands == null || operands.size() < 1) {
+      throw new Pql2CompilationException("Aggregation function expects non null argument");
     }
-    String functionName = function.getOperator();
+
     String columnName;
-    if (functionName.equalsIgnoreCase("count")) {
-      columnName = "*";
+    String functionName = function.getOperator();
+
+    if (functionName.equalsIgnoreCase(AggregationFunctionType.DISTINCT.getName())) {
+      // DISTINCT can support multiple arguments
+      if (operands.size() == 1) {
+        // single column DISTINCT
+        columnName = getColumnExpression(operands.get(0));
```

Comment:
Do we want to support `SELECT DISTINCT * FROM table`?

Reply:
Discussed offline. We don't support that